src/mame/apple/apple2e.cpp

```diff
@@ -4218,7 +4218,7 @@ static INPUT_PORTS_START( apple2e_special )
 	PORT_START("keyb_special")
 	PORT_BIT( 0x01, IP_ACTIVE_LOW,  IPT_KEYBOARD) PORT_NAME("Caps Lock")    PORT_CODE(KEYCODE_CAPSLOCK) PORT_TOGGLE
 	PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_NAME("Left Shift")   PORT_CODE(KEYCODE_LSHIFT)   PORT_CHAR(UCHAR_SHIFT_1)
-	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_NAME("Right Shift")  PORT_CODE(KEYCODE_RSHIFT)
+	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_NAME("Right Shift")  PORT_CODE(KEYCODE_RSHIFT)   PORT_CHAR(UCHAR_SHIFT_1)
```

Comment:
You don’t want to do this – you only want each `UCHAR_SHIFT_n` on a single key.

Reply:
I can make the change.  

Is UCHAR_SHIFT_1 meant for platforms with just 1 shift key?  If so, then should all instances of UCHAR_SHIFT_1 in apple2e.cpp for left and right shifts be replaced with UCHAR_MAMEKEY(LSHIFT) and UCHAR_MAMEKEY(RSHIFT)?

Prior to this PR, UCHAR_SHIFT_1 was used by 7 out of the 10 cases of KEYCODE_RSHIFT.  All 10 cases of KEYCODE_LSHIFT had it.

Reply:
`UCHAR_SHIFT_n` are used to work out which keys to hold to produce a shifted character for UI paste, Lua `emu.keypost` or whatever it’s called, and “natural keyboard” mode.

Consider this line for an Amiga keyboard:
```c++
	PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_A)           PORT_CHAR('a')  PORT_CHAR('A') PORT_CHAR(U'æ') PORT_CHAR(U'Æ')
```

This key has characters for four shift states:
* “a” is produced by pressing this key alone
* “A” is produced by pressing this key while holding the `UCHAR_SHIFT_1` key
* “æ” is produced by pressing this key while holding the `UCHAR_SHIFT_2` key
* “Æ” is produced by pressing this key while holding both the `UCHAR_SHIFT_1` and `UCHAR_SHIFT_2` keys

You need to choose one of the shift keys to be the one that is held to produce shifted characters.  For that Amiga example, Left Shift is `UCHAR_SHIFT_1` and Right Amiga is `UCHAR_SHIFT_2`.

The `UCHAR_MAMEKEY` stuff works a bit strangely.  You can try it in “natural keyboard” mode with `UCHAR_MAMEKEY(RSHIFT)`, but it will produce odd results when using the right shift key if the host keyboard layout doesn’t match the emulated keyboard layout.  I’d just omit the `PORT_CHAR` for the right shift key.

Reply:
My latest commit removes the PORT_CHAR from the right shift keys, including those that had it before this PR.